lib/enmeshed/connector_api.rb

```diff
@@ -0,0 +1,244 @@
+
+    def self.init_conn
+      if User.omniauth_providers.exclude?(:nbp) || CONNECTOR_URL.nil? || API_KEY.nil? || DISPLAY_NAME.nil?
+        raise ConnectorError.new('NBP provider or enmeshed connector not configured as expected')
+      end
+
+      Faraday.new(CONNECTOR_URL, headers:)
+    end
+    private_class_method :init_conn
+
+    def self.enmeshed_address
+      return @enmeshed_address if @enmeshed_address.present?
+
+      identity = parse_result conn.get('/api/v2/Account/IdentityInfo')
+      @enmeshed_address = identity[:address]
+    end
+    private_class_method :enmeshed_address
+
+    def self.display_name_id
+      return @display_name_id if @display_name_id.present?
+
+      new_attribute = parse_result(conn.post('/api/v2/Attributes') do |req|
+        req.body = {content: display_name_attribute}.to_json
+      end)
+
+      @display_name_id = new_attribute[:id]
+    end
+    private_class_method :display_name_id
+
+    def self.headers
+      {'X-API-KEY': API_KEY, 'content-type': 'application/json', accept: 'application/json'}
+    end
+    private_class_method :headers
+  end
+
+  class Relationship
+    def initialize(relationship_json)
+      @json = relationship_json
+      @relationship_changes = relationship_json[:changes] || []
+    end
+
+    attr_reader :json
+
+    def userdata
+      @userdata || parse_userdata
+    end
+
+    def template_expiration
+      DateTime.parse(@json[:template][:expiresAt])
+    end
+
+    def nbp_uid
+      @json.dig(:template, :content, :metadata, :nbp_uid)
+    end
+
+    def id
+      @json[:id]
+    end
+
+    def accept
+      raise ConnectorError('Relationship should exactly one RelationshipChange') if @relationship_changes.size != 1
+
+      Rails.logger.debug do
+        "Enmeshed::ConnectorApi accepting Relationship for template #{@json[:template][:truncatedReference]}"
+      end
+
+      ConnectorApi.respond_to_rel_change(id, @relationship_changes.first[:id], 'Accept')
+    end
+
+    def reject
+      Rails.logger.debug do
+        "Enmeshed::ConnectorApi rejecting Relationship for template #{@json[:template][:truncatedReference]}"
+      end
+
+      @json[:changes].each do |change|
+        ConnectorApi.respond_to_rel_change(id, change[:id], 'Reject')
+      end
+    end
+
+    private
+
+    def parse_userdata # rubocop:disable Metrics/AbcSize
+      # Since the RelationshipTemplate has a `maxNumberOfAllocations` attribute set to 1,
+      # you cannot request multiple Relationships with the same template.
+      # Further, RelationshipChanges should not be possible before accepting the Relationship.
+      raise ConnectorError('Relationship should have exactly one RelationshipChange') if @relationship_changes.size != 1
+
+      change_response_items = @relationship_changes.first.dig(:request, :content, :response, :items)
+
+      read_attribute_accept_response_items = change_response_items.select {|item| item[:@type] == 'ReadAttributeAcceptResponseItem' }
```

Comment:
I tend to use a more descriptive naming (domain wording), e.g. `user_provided_attributes`, to improve understanding.